src/query/dblink_scan.c

```diff
@@ -540,6 +540,64 @@ dblink_bind_param (DBLINK_SCAN_INFO * scan_info, VAL_DESCR * vd, DBLINK_HOST_VAR
   return S_SUCCESS;
 }
 
+int
+dblink_execute_query (struct access_spec_node *spec, VAL_DESCR * vd, DBLINK_HOST_VARS * host_vars)
+{
+  int ret = NO_ERROR, conn_handle, stmt_handle;
+  T_CCI_ERROR err_buf;
+  char conn_url[MAX_LEN_CONNECTION_URL] = { 0, };
+  char *user_name = spec->s.dblink_node.conn_user;
+  char *password = spec->s.dblink_node.conn_password;
+  char *sql_text = spec->s.dblink_node.conn_sql;
+
+  char *find = strstr (spec->s.dblink_node.conn_url, ":?");
+  if (find)
+    {
+      snprintf (conn_url, MAX_LEN_CONNECTION_URL, "%s%s", spec->s.dblink_node.conn_url, "&__gateway=true");
+    }
+  else
+    {
+      snprintf (conn_url, MAX_LEN_CONNECTION_URL, "%s%s", spec->s.dblink_node.conn_url, "?__gateway=true");
+    }
+
+  conn_handle = cci_connect_with_url_ex (conn_url, user_name, password, &err_buf);
+  if (conn_handle < 0)
+    {
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_DBLINK, 1, err_buf.err_msg);
+      goto error_exit;
+    }
+  else
+    {
+      cci_set_autocommit (conn_handle, CCI_AUTOCOMMIT_TRUE);
+      stmt_handle = cci_prepare (conn_handle, sql_text, 0, &err_buf);
+      if (stmt_handle < 0)
+	{
+	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_DBLINK, 1, err_buf.err_msg);
+	  goto error_exit;
+	}
+
+      if (host_vars->count > 0)
+	{
+	  if ((ret = dblink_bind_param (stmt_handle, vd, host_vars)) < 0)
+	    {
+	      return ret;
+	    }
+	}
+
+      ret = cci_execute (stmt_handle, 0, 0, &err_buf);
+      if (ret < 0)
+	{
+	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_DBLINK, 1, err_buf.err_msg);
+	  goto error_exit;
+	}
+    }
```

Comment:
After the end of this function, the conn_handle value can not be referenced. Is it ok not to disconnect it?

Reply:
Good point. I will fix it.